src/Expr/ClosureExpressionVisitor.php

```diff
@@ -122,24 +122,24 @@ public function walkComparison(Comparison $comparison): Closure
         $value = $comparison->getValue()->getValue();
 
         return match ($comparison->getOperator()) {
-            Comparison::EQ => static fn ($object): bool => self::getObjectFieldValue($object, $field) === $value,
-            Comparison::NEQ => static fn ($object): bool => self::getObjectFieldValue($object, $field) !== $value,
-            Comparison::LT => static fn ($object): bool => self::getObjectFieldValue($object, $field) < $value,
-            Comparison::LTE => static fn ($object): bool => self::getObjectFieldValue($object, $field) <= $value,
-            Comparison::GT => static fn ($object): bool => self::getObjectFieldValue($object, $field) > $value,
-            Comparison::GTE => static fn ($object): bool => self::getObjectFieldValue($object, $field) >= $value,
-            Comparison::IN => static function ($object) use ($field, $value): bool {
+            Comparison::EQ => static fn (object|array $object): bool => self::getObjectFieldValue($object, $field) === $value,
```

Comment:
This test is the reason why I used `array|object` here and below: https://github.com/doctrine/collections/blob/16dd2967fa326bb8c207e400e448a501b81e2eb8/tests/ClosureExpressionVisitorTest.php#L415-L421